### PR TITLE
examples: added monaco-react example

### PR DESCRIPTION
**Pull request by mertcanaltin**
added monaco-react example
[issue](https://github.com/vercel/next.js/issues/25263)

### CODE REVIEW

**Comment by dvakatsiienko** [comment]
the **Socket Security: Pull Request Alerts** step check fails most likely due to existing changes in **package.json**. but in your case it is entirely new file. But since it exists in PR files changes, it still being checked by that step check.
#49997

**Review by leerob — CHANGES_REQUESTED** [review]
Could this be updated to use the App Router and follow the same README format as other examples? Thank you.

**Comment by mertcanaltin** [comment]
> Could this be updated to use the App Router and follow the same README format as other examples? Thank you.

I made an update 🚀

File: examples/monaco-react/package.json
Comment (leerob) [inline examples/monaco-react/package.json:3]:
```suggestion
```

File: examples/monaco-react/package.json
Comment (leerob) [inline examples/monaco-react/package.json:20]:
```suggestion
    "next": "latest",
```

File: examples/monaco-react/pages/index.tsx
Comment (leerob) [inline examples/monaco-react/pages/index.tsx:3]: This should be `app/page.tsx` and you'll also need a root layout `app/layout.tsx`.

https://nextjs.org/docs/getting-started/installation

File: examples/next-monaco/tsconfig.json
Comment (leerob) [inline examples/next-monaco/tsconfig.json:2]: Could you remove the duplicates here? There should be one folder, `monaco-react`.

Comment (mertcanaltin) [reply examples/next-monaco/tsconfig.json:2]: is it this way @leerob 

```ts
{
  "exclude": ["node_modules", "monaco-react"]
}
```

Comment (mertcanaltin) [reply examples/next-monaco/tsconfig.json:2]: > I mean that there's two examples, two folders, in this PR. We should just have one 😄

sorry i missed this, i'll fix it now

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: ec2fdd797a07b7bfcd5a3642c519d0b64c0f04f4

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: ec2fdd797a07b7bfcd5a3642c519d0b64c0f04f4

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ppbl** [comment]
Can you add cases that do not use CDN? The default CDN seems to be unavailable in some regions

There seems to be still a problem with `loader.config({ monaco })`

**Comment by ppbl** [comment]
Local packaging has been resolved and the filename field needs to be configured
 ```js
config.plugins.push(
        new MonacoWebpackPlugin({
          languages: ['json'],
          filename: 'static/[name].worker.js',
        })
)```

**Comment by mertcanaltin** [comment]
hello i need help here i wonder where should i write this config

**Comment by mertcanaltin** [comment]
Hello, is this okey? @leerob

**Comment by remcohaszing** [comment]
While I appreciate the addition of an example for using Monaco editor with Next.js, I don’t think this is the right solution. Third party React integrations tend to abstract away [Monaco’s core concepts](https://github.com/microsoft/monaco-editor#concepts) and present it as a simple text editor instead.

The reason this specific library works, is because it loads Monaco from a CDN. The biggest reason people are having trouble loading the `monaco-editor` and `monaco-editor-core` libraries in Next.js, is because those import CSS files. Next.js blocks this for unknown reasons.

See https://github.com/vercel/next.js/issues/34501

**Comment by leerob** [comment]
The App Router now enables Next.js to more easily import CSS files from third-party packages. Does monaco publish those files separately in the package to allow frameworks to import them?

**Comment by remcohaszing** [comment]
No. Files from `monaco-editor` directly import CSS files. For example, see `esm/vs/base/browser/ui/countBadge/countBadge.js` on https://www.npmjs.com/package/monaco-editor/v/0.45.0?activeTab=code

File: examples/next-monaco-editor/package.json
Comment (giraffesyo) [inline examples/next-monaco-editor/package.json:21]: Why does this have both `react-monaco-editor` and `@monaco-editor/react`?

**Review by giraffesyo — CHANGES_REQUESTED** [review]

**Comment by mertcanaltin** [comment]
> While I appreciate the addition of an example for using Monaco editor with Next.js, I don’t think this is the right solution. Third party React integrations tend to abstract away [Monaco’s core concepts](https://github.com/microsoft/monaco-editor#concepts) and present it as a simple text editor instead.
> 
> The reason this specific library works, is because it loads Monaco from a CDN. The biggest reason people are having trouble loading the `monaco-editor` and `monaco-editor-core` libraries in Next.js, is because those import CSS files. Next.js blocks this for unknown reasons.
> 
> See #34501

@leerob I wonder if you agree with this, I want to update and advance this pull request, do you think I should close it, thanks again for your support

**Comment by giraffesyo** [comment]
I also agree that @monaco-editor/react is not the right solution, at least with the provided configuration in this PR. As it stands, this loads the monaco-editor files from the jsdelivr CDN, which is not appropriate in many use cases. I believe the previous example in the Next repo was working fully locally, until it broke in a later version of Next.js (11?).

**Comment by mertcanaltin** [comment]
> I also agree that @monaco-editor/react is not the right solution, at least with the provided configuration in this PR. As it stands, this loads the monaco-editor files from the jsdelivr CDN, which is not appropriate in many use cases. I believe the previous example in the Next repo was working fully locally, until it broke in a later version of Next.js (11?).  

this is an old pull request I agree with what you said we can improve our structure here if you have a suggestion please do not hesitate

**Comment by leerob** [comment]
Cross posting here a solution: https://github.com/remcohaszing/monaco-yaml/issues/219#issuecomment-2198118949

**Pull request closed without merging**
